packages/sil-governance/src/routes/create-proposal/components/ProposalForm.tsx

```diff
@@ -1,53 +1,75 @@
-import { Block, Button, Form, SelectFieldFormItem, Typography, useForm } from "medulas-react-components";
+import { ProposalOptions, ProposalType } from "@iov/bns-governance";
+import { Block, Button, Form, FormValues, Typography, useForm } from "medulas-react-components";
 import React, { useState } from "react";
+import * as ReactRedux from "react-redux";
 
+import { sendSignAndPostRequest } from "../../../communication/signandpost";
+import { getBnsConnection } from "../../../logic/connection";
+import { RootState } from "../../../store/reducers";
 import DescriptionField from "./DescriptionField";
+import FormOptions from "./FormOptions";
 import ParticipationData from "./ParticipationData";
-import FormOptions from "./ProposalOptions";
-import ProposalTypeField from "./ProposalTypeField";
+import ProposalTypeSelect from "./ProposalTypeSelect";
 import TitleField from "./TitleField";
 import WhenField from "./WhenField";
 
-const formOptions = {
-  addCommitteeMember: "Add Committee Member",
-  removeCommitteeMember: "Remove Committee Member",
-  amendCommitteeThreshold: "Amend Committee Threshold",
-  amendCommitteeQuorum: "Amend Committee Quorum",
-  releaseGuaranteeFunds: "Release Guarantee Funds",
-  distributeFunds: "Distribute Funds",
-  addValidator: "Add Validator",
-  removeValidator: "Remove Validator",
-  amendProtocol: "Amend Blockchain Software",
-};
+const buildProposalOptions = (type: ProposalType, values: FormValues): ProposalOptions => {
+  console.log(JSON.stringify(values));
+
+  const [year, month, day] = values["Date"].split("-").map(el => parseInt(el));
+  const [hour, minute] = values["Time"].split(":").map(el => parseInt(el));
 
-const onSubmit = (): void => {};
+  const title = values["Title"];
+  const description = values["Description"];
+  const startTime = new Date(year, month - 1, day, hour, minute);
+  //TODO Get Election Rule from values when the UI for selecting committee is done
+  const electionRuleId = 1;
+
+  const commonOptions = {
+    title,
+    description,
+    startTime,
+    electionRuleId,
+  };
+
+  const text = values["Text"];
+
+  switch (type) {
+    default:
+      return { ...commonOptions, type: ProposalType.AmendProtocol, text };
```

Comment:
The default should probably be to throw. This `return` statement can be in a `case ProposalType.AmendProtocol`.